### PR TITLE
Connect Inputs and Outputs Properly and Export CopyColumnTransform to (unofficial) ONNX operator

### PR DESCRIPTION
**Pull request by wschin**
Related issue: #955 , #968.
1. Add exporter for CopyColumnTransform. It only creates a ONNX node with a `type` field for encoding the transform's C# name.
2. Properly connect inputs and outputs especially when they have the same name in pipeline.

### CODE REVIEW

**Comment by wschin** [comment]
@dotnet-bot maybe can rebuild?

File: src/Microsoft.ML.Data/Model/Onnx/OnnxContext.cs [hunk 1]
```diff
@@ -23,6 +23,13 @@ public abstract class OnnxContext
         /// <returns>A name that has not yet been returned from this function, starting with <paramref name="prefix"/></returns>
         public abstract string GetNodeName(string prefix);
 
+        /// <summary>
+        /// Determine if a string has been used as ONNX variable name somewhere.
+        /// </summary>
+        /// <param name="variableName">examined string</param>
+        /// <returns>True if the input argument has been used to denote an ONNX variable. Otherwise, False.</returns>
+        public abstract bool IsDefined(string variableName);
```
Comment (codemzs) [inline src/Microsoft.ML.Data/Model/Onnx/OnnxContext.cs]: >IsDefined [](start = 29, length = 9)

Can we please rename this to be more specific like "IsVariableDefined"?

File: src/Microsoft.ML.Onnx/SaveOnnxCommand.cs [hunk 2]
```diff
@@ -232,13 +232,15 @@ private void Run(IChannel ch)
                 if (end.Schema.IsHidden(i))
                     continue;
 
-                var idataviewColumnName = end.Schema.GetColumnName(i);;
-                if (_outputsToDrop.Contains(idataviewColumnName) || _inputsToDrop.Contains(idataviewColumnName))
```
Comment (codemzs) [inline src/Microsoft.ML.Onnx/SaveOnnxCommand.cs]: > if (_outputsToDrop.Contains(idataviewColumnName) || _inputsToDrop.Contains(idataviewColumnName)) [](start = 15, length = 97)

Why did you remove inputsToDrop?

File: src/Microsoft.ML.Data/Model/Onnx/OnnxContext.cs (hunk 1 above)
Comment (wschin) [reply src/Microsoft.ML.Data/Model/Onnx/OnnxContext.cs]: Sure.

---
In reply to: [222023194](https://github.com/dotnet/machinelearning/pull/952#discussion_r222023194) [](ancestors = 222023194)

File: src/Microsoft.ML.Onnx/SaveOnnxCommand.cs (hunk 2 above)
Comment (wschin) [reply src/Microsoft.ML.Onnx/SaveOnnxCommand.cs]: We can't drop variables without considering the whole graph structure. For each non-hidden variable in output schema, it must either be computed somewhere or just an input. For the first case, we don't need to remove it. For the second case, we should connect that an input to that output via Identity.

---
In reply to: [222025561](https://github.com/dotnet/machinelearning/pull/952#discussion_r222025561) [](ancestors = 222025561)

File: src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs [hunk 3]
```diff
@@ -197,6 +200,21 @@ public override RowMapperColumnInfo[] GetOutputColumns()
                 }
                 return result;
             }
+
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                var infos = GetOutputColumns();
```
Comment (vaeksare) [inline src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs]: You are not using this infos anywhere. I am guessing it was originally meant to be used in the foreach but you decided to use _columns instead?
